noise-research/secure-transfer.md

```diff
@@ -0,0 +1,243 @@
+    - obtains from decrypting the received message a public key `sA`. If `sA` is not a valid public key, the protocol is aborted.
+    - performs `DH(sA,eB)` (which updates a symmetric encryption key);
+    - performs `DH(sA,sB)` (which updates a symmetric encryption key);
+    - decrypts the payload to obtain the randomness `s`. 
+    - Computes `H(sA||s)` and checks if this value corresponds to the commitment obtained in step 6. If not, the protocol is aborted.
+    - Calls Split() and obtains two cipher states to encrypt inbound and outbound messages.
+
+### The `WakuPairing` for Devices without a Camera
+
+In the above pairing handshake, the QR is by default exposed by device `B` and not by `A` because device `B` locally stores no relevant cryptographic material, so an active local attacker that scans the QR code first would only be able to transfer *his own* session information and get nothing from `A`. 
+
+However, since the user confirms at the end of message `1` that the authorization code is the same on both devices, the role of handhsake initiator and responder can be safely swapped in message `0` and `1`. 
+
+This allows pairing in case device `A` does not have a camera to scan a QR (e.g. a desktop client) while device `B` has.
+
+The resulting handshake would then be:
+```
+WakuPairing2:
+0.   -> eA              {H(sB||r), contentTopicParams, messageNametag}
+     ...
+1.   <- eB, eAeB        {H(sB||r)}   [authcode]
+2.   <- sB, eAsB        {r}
+3.   -> sA, sAeB, sAsB  {s}
+
+{}: payload,    []: user interaction
+```
+
+## Security Analysis
+
+### Assumptions
+
+- The attacker is active, i.e. can interact with both devices `A` and `B` by sending messages over `contentTopic`.
+
+- The attacker has access to the QR code, that is knows the ephemeral key `eB`, the commitment `H(sB||r)` and the `contentTopic` exposed by the device `B`.
+
+- Devices `A` and `B` are considered trusted (otherwise the attacker will simply exfiltrate the relevant information from the attacked device).
+
+- As common for Noise, we assume that ephemeral keys cannot be compromised, while static keys might be later compromised. However, we enforce in the pairing phase extra security mechanisms (i.e. use of commitments for static keys) that will prevent some attacks possible when ephemeral keys are weak or get compromised.
+ 
+### Rationale
+
+- The device `B` exposes a commitment to its static key `sB` because:
+    - if the private key of `eB` is weak or gets compromised, an attacker can impersonate `B` by sending in message `2` to device `A` his own static key and successfully complete the pairing phase. Note that being able to compromise `eB` is not contemplated by our security assumptions.
+    - `B` cannot adaptively choose a static key based on the state of the Noise handshake at the end of message `1`, i.e. after the authentication code is confirmed. Note that device `B` is trusted in our security assumptions.
```

Comment:
How B can benefit from such an adaptive selection of its static key? or do you mean an Attacker? And if B is already trusted, then I'd say there is no need to prevent such an attack (although the attack is yet not clear to me)